rDNA/R/rDNA.R

```diff
@@ -1806,4 +1807,289 @@ print.dna_barplot <- function(x, trim = 30, attr = TRUE, ...) {
+                                             y = "pos",
+                                             label = "label"),
+                         size = (fontSize / ggplot2::.pt),
+                         inherit.aes = FALSE,
+                         data = object2)
+  } else if (max(w) > 0 & min(w) < 0) {
+    g <- g +
+      ggplot2::geom_text(ggplot2::aes_string(color = "text_color"),
+                         size = (fontSize / ggplot2::.pt),
+                         position = ggplot2::position_stack(vjust = 0.5, reverse = TRUE),
+                         inherit.aes = TRUE,
+                         data = object2[object2$frequency >= 0,]) +
+      ggplot2::geom_text(ggplot2::aes_string(color = "text_color"),
+                         size = (fontSize / ggplot2::.pt),
+                         position = ggplot2::position_stack(vjust = 0.5),
+                         inherit.aes = TRUE,
+                         data = object2[object2$frequency < 0,])
+  } else if (min(w) >= 0) {
+    g <- g +
+      ggplot2::geom_text(ggplot2::aes_string(color = "text_color"),
+                         size = (fontSize / ggplot2::.pt),
+                         position = ggplot2::position_stack(vjust = 0.5, reverse = TRUE),
+                         inherit.aes = TRUE)
+  } else {
+    g <- g +
+      ggplot2::geom_text(ggplot2::aes_string(color = "text_color"),
+                         size = (fontSize / ggplot2::.pt),
+                         position = ggplot2::position_stack(vjust = 0.5),
+                         inherit.aes = TRUE)
+  }
+  if (lab) {
+    g <- g +
+      ggplot2::annotate("text",
+                        x = high,
+                        y = offset * 2,
+                        hjust = 0,
+                        label = lab.pos,
+                        size = (fontSize / ggplot2::.pt)) +
+      ggplot2::annotate("text",
+                        x = high,
+                        y = 0 - offset * 2,
+                        hjust = 1,
+                        label = lab.neg,
+                        size = (fontSize / ggplot2::.pt)) +
+      ggplot2::scale_x_discrete(labels = function(x) trim(x, n = truncate),
+                                expand = c(0, 2, 0, 2),
+                                limits = levels(object2$entity))
+  } else {
+    g <- g +
+      ggplot2::scale_x_discrete(labels = function(x) trim(x, n = truncate),
+                                limits = levels(object2$entity))
+  }
+  return(g)
+}
+
+#' Truncate labels
+#'
+#' Internal function, used to truncate labels.
+#'
+#' @param x A character string
+#' @param n Max number of characters to truncate to. Value \code{Inf} turns off
+#'   truncation.
+#' @param e String added at the end of x to signal it was truncated.
+#'
+#' @noRd
+#'
+#' @author Johannes B. Gruber
+trim <- function(x, n, e = "...") {
+  ifelse(nchar(x) > n,
+         paste0(gsub("\\s+$", "",
+                     strtrim(x, width = n)),
+                e),
+         x)
```

Comment:
Can you make sure this function works in the same way as the trimming in `print.dna_barplot` and other `print` methods (i.e., the output looks trimmed in the same way? Ideally, I would think this can be done inside the function itself and doesn't require a helper function like this, just like I did in those `print` methods -- can you change it accordingly?

Reply:
Ok sure, I'll kick out the `trim` function. Just thought it might be more effective to have an additional helper function, rather than defining the trimming inside every new function.

Reply:
It's a single line of code to do the trimming. So it shouldn't matter much if it's in a helper or done directly. This way, we can at least remove one layer of code complexity.